File: examples/example-kmp/composeApp/src/androidMain/kotlin/app/screeb/example/kmp/MainActivity.kt
package app.screeb.example.kmp

import android.content.Intent
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import app.screeb.sdk.kmp.Screeb

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            App(channelId = "0e2b609a-8dce-4695-a80f-966fbfa87a88")
        }
        // Forward the launch deep link (e.g. screeb-<channel-id>://inspector) to the SDK.
        Screeb.handleDeepLink(intent?.data?.toString())
    }

    override fun onNewIntent(intent: Intent) {
        super.onNewIntent(intent)
        setIntent(intent)
        Screeb.handleDeepLink(intent.data?.toString())
    }
}

File: examples/example-maui/Platforms/Android/MainActivity.cs
using Android.App;
using Android.Content;
using Android.Content.PM;
using Android.OS;

namespace ExampleMaui;

[Activity(
    Theme = "@style/Maui.SplashTheme",
    MainLauncher = true,
    LaunchMode = LaunchMode.SingleTop,
    ConfigurationChanges = ConfigChanges.ScreenSize | ConfigChanges.Orientation |
                           ConfigChanges.UiMode | ConfigChanges.ScreenLayout |
                           ConfigChanges.SmallestScreenSize | ConfigChanges.Density)]
// Screeb deep links: screeb-<channel-id>://... (e.g. ://inspector)
[IntentFilter(
    new[] { Intent.ActionView },
    Categories = new[] { Intent.CategoryDefault, Intent.CategoryBrowsable },
    DataScheme = "screeb-0e2b609a-8dce-4695-a80f-966fbfa87a88")]
public class MainActivity : MauiAppCompatActivity
{
    protected override void OnCreate(Bundle? savedInstanceState)
    {
        base.OnCreate(savedInstanceState);
        // Forward the launch deep link to the SDK.
        global::App.Screeb.Sdk.Screeb.Instance.HandleDeepLink(Intent);
    }

    protected override void OnNewIntent(Intent? intent)
    {
        base.OnNewIntent(intent);
        Intent = intent;
        global::App.Screeb.Sdk.Screeb.Instance.HandleDeepLink(intent);
    }
}
